### ISSUE TITLE
consider using std::span

### ISSUE THREAD

**Issue by lemire** [oschonrock/binfuse#1]
Not everyone uses `std::vector`. Something more generic like `std::span` might be useful:

```C++

void populate(std::span<const std::uint64_t> keys) {
    if (keys.empty()) {
        throw std::runtime_error("empty input");
    }
    size = keys.size();

    if (!ftype<FilterType>::allocate(keys.size(), &fil)) {
        throw std::runtime_error("failed to allocate memory.\n");
    }
    if (!ftype<FilterType>::populate(
            keys.data(), 
            keys.size(), &fil)) {
        throw std::runtime_error("failed to populate the filter");
    }
}
```

(I have not tested this code.)

**Comment by oschonrock** [comment]
yes, good thought... anything else you think of, just note it down

**Comment by lemire** [comment]
@oschonrock I am not super excited about `estimate_false_positive_rate()`. I would put it outside the class definition. It is a possible trap as it is going to be slightly expensive. It is really running a simulation.

If you want to leave it in the class, I would document what it does... E.g., this should not be called often.

**Comment by oschonrock** [comment]
> If you want to leave it in the class, I would document what it does... E.g., this should not be called often.

yes, fair point, I had similar thoughts.. It almost belongs in the tests not in the class.

**Comment by oschonrock** [comment]
ok, those are done in https://github.com/oschonrock/binfuse/commit/5a1bede895fbf93c8088a352ad583b591c786c45

going to close this issue and open another one with specific API questions/doubt that I have.. Here it is:

https://github.com/oschonrock/binfuse/issues/2
